src/graphql/schema/collection/createCollectionGqlType.ts

```diff
@@ -45,12 +46,17 @@ export default function createCollectionGqlType<TValue> (
       Array.from(type.fields).map(
         <TFieldValue>([fieldName, field]: [string, ObjectType.Field<TValue, TFieldValue>]) => {
           const { gqlType, intoGqlOutput } = getGqlOutputType(buildToken, field.type)
+          const formattedFieldName = formatName.field(fieldName)
           return {
-            key: formatName.field(fieldName),
+            key: formattedFieldName
             value: {
               description: field.description,
               type: gqlType,
-              resolve: (value: TValue): mixed => intoGqlOutput(field.getValue(value)),
+              sqlName: () => fieldName,
```

Comment:
In my initial design for PostGraphQL, the GraphQL and SQL bits were kept completely separate. I’d like to see if we can maintain that pattern as it would make it easier to support other noSQL backends in the future. If it proves too difficult then we can scrap that ideal, but I’d like to try 😊

Here specifically we have `sqlName` and `sqlExpression` in the GraphQL type creation code, and we are also importing the SQL utility from the Postgres package. I’m going to briefly lay out how I saw this working, and we can discuss how to go from there.

I thought that instead of having fields like `sqlName` or `sqlExpression` that we could instead have a `dependencies` field which takes an array of objects from `src/interface`. So for example, the `dependencies` property could have an [`ObjectType.Field`](https://github.com/calebmer/postgraphql/blob/b54d05acafa8ada1d27e1a0e2a4cac091c620fe3/src/interface/type/ObjectType.ts#L34) item, or a [`CollectionKey`](https://github.com/calebmer/postgraphql/blob/b54d05acafa8ada1d27e1a0e2a4cac091c620fe3/src/interface/collection/CollectionKey.ts) item (for node ids). GraphQL would not know anything about these values, but there Postgres implementations (like [`PgObjectType.Field`](https://github.com/calebmer/postgraphql/blob/b54d05acafa8ada1d27e1a0e2a4cac091c620fe3/src/postgres/inventory/type/PgClassType.ts#L110)) would have the SQL information necessary to only select an optimal subset of information.

We could then also add to `src/interface` a `ReadOptimizations` type which would look something like:

```ts
interface ReadOptimizations {
  dependencies: {
    fields: Array<ObjectType.Field>,
    collectionKeys: Array<CollectionKey>,
    ...
  },
}
```

A `ReadOptimizations` object would be generated in `src/graphql` by looking at `resolveInfo` and the GraphQL type. `ReadOptimizations` would then be passed to the interface read methods instead of `resolveInfo` where Postgres could use the information to generate an optimized query.

Admittedly, I was not thinking about how we could optimize procedures when I first thought up this abstraction, but it shouldn’t be too hard, I think. We would just add a private, optional, `_pgProcedures` property to `ReadOptimizations.dependencies` that our Postgres implementation could pick up.

What do you think?